### ISSUE TITLE
Cythonize regression for the combination of @property and generator

### ISSUE THREAD

**Issue by bjodah** [cython/cython#1447]
```cython
cdef class MyClass(object):
    @property
    def flat(self):
        yield 42
```
I bisected the regression to  9b045609cd976c1479b6f7f9fda6c742875a83ee
@linkmauve

**Comment by robertwb** [comment]
Thanks for tracking this down.

**Comment by robertwb** [comment]
Fixed at 3c9326ce4741bcef9c644e46927b1debc83a410a
